Makefile

```diff
@@ -40,12 +52,22 @@ help:
 	@echo '  usig-*           - Make USIG target, where target is one of:'
 	@echo '                     $(usig-target-list)'
 
-build: usig-build | bin
-	go build -o sample/bin/peer ./sample/peer
-	go build -o sample/bin/keytool ./sample/authentication/keytool
+build: usig-build
+	go build -o $(builddir)/keytool ./sample/authentication/keytool
+	go build -o $(builddir)/peer ./sample/peer
+
+install: build
```

Comment:
Personally I don't think that target `install` has to depend on target `build` because `make uninstall` had better be a reverse operation of `make install`. Or if you like to keep this dependency, maybe the help message of `make install` should be "Build and install artifacts".

Reply:
Although it might sound a bit unlogical, but it is actually a convention that `install` target builds first: https://www.gnu.org/prep/standards/html_node/Standard-Targets.html.

Reply:
I looked a Makefile generated by famous generators: [Automake](https://www.gnu.org/software/automake/), [CMake](https://www.gnu.org/software/automake/) and find that it follows the convention mentioned by @sergefdrv (CMake does not generate `uninstall` target [automatically](https://gitlab.kitware.com/cmake/community/wikis/FAQ#can-i-do-make-uninstall-with-cmake)). I would like to keep this code to follow the convention. Thanks for insightful review @Naoya-Horiguchi, and comments @sergefdrv.